zcash_client_sqlite/src/error.rs

```diff
@@ -33,15 +34,16 @@ pub enum SqliteClientError {
     /// Illegal attempt to reinitialize an already-initialized wallet database.
     TableNotEmpty,
 
-    /// A Bech32-encoded key or address decoding error
-    Bech32DecodeError(Bech32DecodeError),
+    /// A Zcash key or address decoding error
+    DecodingError(ParseError),
 
     /// An error produced in legacy transparent address derivation
     #[cfg(feature = "transparent-inputs")]
     HdwalletError(hdwallet::error::Error),
 
     /// An error encountered in decoding a transparent address from its
     /// serialized form.
+    #[cfg(feature = "transparent-inputs")]
```

Comment:
Why is this specific to transparent-inputs? It seems as though it could also occur in principle due to support for transparent outputs, and the fact that it doesn't currently is just an implementation detail.